numpy/ma/core.py

```diff
@@ -3184,13 +3180,15 @@ def __setitem__(self, indx, value):
 
         """
         # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
+        if (self._mask is not nomask) and \
+                (self._sharedmask or not self._data.flags.owndata):
```

Comment:
Not to mention that if you mask an existing array the data is a view, so that won't work. Reference counting would be a source of endless troubles. Hmm...